pallets/circuit/src/lib.rs

```diff
@@ -329,19 +357,63 @@ pub mod pallet {
         // This function must return the weight consumed by `on_initialize` and `on_finalize`.
         fn on_initialize(n: T::BlockNumber) -> Weight {
             let weight = Self::process_signal_queue();
```

Comment:
We do need to add the ticket for handling multiple queue management scenarios here:

- overusing blocks, too much consumption from any of these iterations
- taking big iterables, need to be able to take N for each and if we reach our consumption, run on the subsequent N blocks until we are back to normal running

Basically we need to be able to manage dead letter queueing especially if we don't have enough time to complete all of these loops in a block, otherwise we risk bids being left, timeouts everywhere being powned.